libraries/AP_Eigen/AP_Eigen.h

```diff
@@ -0,0 +1,35 @@
+#ifndef AP_EIGEN_H
+#define AP_EIGEN_H
+
+/*
+ * Eigen interface extensions
+ */
+#define EIGEN_MATRIXBASE_PLUGIN <AP_Eigen/AP_BaseAddons.h>
+#define EIGEN_MATRIX_PLUGIN     <AP_Eigen/AP_MatrixAddons.h>
+
+/*
+ * The PX4 stack creates a lot of drama if just using C99 math features :D
+ */
+#if CONFIG_HAL_BOARD == HAL_BOARD_PX4 && CONFIG_HAL_BOARD != HAL_BOARD_LINUX // maybe more?!
```

Comment:
if CONFIG_HAL_BOARD is HAL_BOARD_PX4 it's always != CONFIG_HAL_LINUX. Maybe you meant an || rather than &&?